api/app/api.py

```diff
@@ -20,6 +30,62 @@
 )
 
 @app.post("/forecast/")
-async def forecast(site:PVSite):
-    df =run_forecast(site)
-    return df.to_dict()
\ No newline at end of file
+def forecast(request: ForecastRequest):
+    site = request.site
+    ts = request.timestamp if request.timestamp else datetime.now(timezone.utc).isoformat()
+    nwp_source = request.nwp_source
+    access_token = request.access_token
+    enphase_system_id = request.enphase_system_id
+
+    timestamp = pd.Timestamp(ts).tz_localize(None)
+    formatted_timestamp = timestamp.strftime('%Y-%m-%d %H:%M:%S')
+
+    # Run forecast without live data
+    site_no_live = PVSite(latitude=site.latitude, longitude=site.longitude, capacity_kwp=site.capacity_kwp)
+    predictions_no_live = run_forecast(site=site_no_live, ts=timestamp, nwp_source=nwp_source)
+
+    if site.inverter_type == 'enphase' and access_token and enphase_system_id:
```

Comment:
I think the first check should potentially be if there is an inverter type at all, if it's None you can just run the regular forecast and just return that no need for the live and no live columns

Reply:
> I think the first check should potentially be if there is an inverter type at all, if it's None you can just run the regular forecast and just return that no need for the live and no live columns

Don't we know that from the checks? No Inverter is also a type of inverter, but it doesn't exist in the pvsite class , so I treats it as no inverter

Reply:
I would look at the else statement you have at the moment again, it duplicates data at the moment for the case with no inverter which will most likely be the most common type of use of this API I imagine

Reply:
> I would look at the else statement you have at the moment again, it duplicates data at the moment for the case with no inverter which will most likely be the most common type of use of this API I imagine

Do I keep a separate else condition for no inverter?

Reply:
> I would look at the else statement you have at the moment again, it duplicates data at the moment for the case with no inverter which will most likely be the most common type of use of this API I imagine

ive made this change @Sukh-P